contracts/modules/interest-rate-models/IRMClassLido.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "../../BaseIRM.sol";
+
+interface ILidoOracle {
+    function getLastCompletedReportDelta() external view returns (uint postTotalPooledEther, uint preTotalPooledEther, uint timeElapsed);
+}
+
+contract IRMClassLido is BaseIRM {
+    uint constant A_DAY = 24 * 60 * 60;
+    address public immutable lidoOracle;
+    uint public immutable slope1;
+    uint public immutable slope2;
+    uint public immutable kink;
+
+    struct IRMLidoStorage {
+        int96 baseRate;
+        uint64 lastCalled;
+    }
+
+    constructor(bytes32 moduleGitCommit_) BaseIRM(MODULEID__IRM_CLASS__LIDO, moduleGitCommit_) {
+        lidoOracle = 0x442af784A788A5bd6F42A01Ebe9F287a871243fb;
+
+        // Base=Lido APY,  Kink(80%)=8% APY  Max=200% APY
+        slope1 = 709783723;
+        slope2 = 37689273223;
+        kink = 3435973836;
+    }
+
+    function computeInterestRateImpl(address, uint32 utilisation) internal override returns (int96) {
+        uint ir = 0;
+        if (utilisation > 0) {
+            IRMLidoStorage storage irmLido;
+            {
+                bytes32 storagePosition = keccak256("euler.irm.class.lido");
+                assembly { irmLido.slot := storagePosition }
+            }
+
+            if (block.timestamp - irmLido.lastCalled > A_DAY) {
+                (bool success, bytes memory data) = lidoOracle.staticcall(abi.encodeWithSelector(ILidoOracle.getLastCompletedReportDelta.selector));
+                
+                // if the Lido oracle call unsuccessful, the base rate will be set to the last stored value
+                if (success) {
```

Comment:
Might also want to ensure that `data.length >= (32*3)` here, for sanity checking, otherwise the `abi.decode` could fail.